packages/domain/lib/usecase/get_movie_response_usecase.dart

```diff
@@ -0,0 +1,60 @@
+import '../enum/movie_type.dart';
+import '../utils/const.dart';
+import '../model/movie_response_model.dart';
+import '../repository/movie_repository.dart';
+import 'usecase.dart';
+
+class GetMovieResponseUseCase
+    implements UseCaseParams<MovieType, Future<List<MovieResponse>>> {
+  final NetworkRepository _repository;
+
+  GetMovieResponseUseCase(this._repository);
+
+  @override
+  Future<List<MovieResponse>> call(movieType) async {
```

Comment:
1) looks top!!
2) add type to movieType param
3) move queryParams forming into a separate method (now u have a code duplication)